frontend/src/pages/journal/journal.tsx

```diff
@@ -1,16 +1,47 @@
 import { BackButton } from '#libs/components/components.js';
+import { AppRoute } from '#libs/enums/enums.js';
 import { getValidClassNames } from '#libs/helpers/helpers.js';
-import { useCallback, useSidebarState } from '#libs/hooks/hooks.js';
+import {
+  useAppDispatch,
+  useAppSelector,
+  useCallback,
+  useEffect,
+  useNavigate,
+  useParams,
+  useSidebarState,
+} from '#libs/hooks/hooks.js';
+import { actions as journalActions } from '#slices/journal/journal.js';
 
 import { JournalSidebar } from './components/journal-sidebar/journal-sidebar.js';
+import { Note } from './components/note/note.js';
 import styles from './styles.module.scss';
 
 const Journal: React.FC = () => {
+  const { id } = useParams();
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
   const { isSidebarShown, setIsSidebarShown } = useSidebarState();
 
+  const { selectedJournalEntry } = useAppSelector(({ journal }) => {
+    return {
+      selectedJournalEntry: journal.selectedJournalEntry,
+    };
+  });
+
+  const hasNoteId = Boolean(id) && selectedJournalEntry;
+
   const handleBackButtonPress = useCallback(() => {
+    navigate(AppRoute.JOURNAL);
     setIsSidebarShown(true);
-  }, [setIsSidebarShown]);
+  }, [setIsSidebarShown, navigate]);
+
+  useEffect(() => {
+    void dispatch(journalActions.getAllJournalEntries()).then(() => {
+      if (id) {
+        void dispatch(journalActions.setSelectedJournalEntry(Number(id)));
+      }
+    });
```

Comment:
can't we use await syntax, maybe without void?

Reply:
I can change it to this. Is this ok?
![image](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/102380564/aae688b9-2f1b-478e-be62-199fca7674f3)

Reply:
can't we just split it into two useEffects? looks like we don't need to await all journal entries to set selected journal entry

Reply:
No, we need to wait for the getAllJournalEntries fulfilled to select an item. And if it splits to 2 effect it will cause infinity sending setAllJournalEntries

Reply:
ok, then let's use as on screenshot above